cli/tests/test_global_opts.rs

```diff
@@ -53,6 +53,10 @@ fn test_version() {
     let expected = [
         "jj ?.??.?\n",
         "jj ?.??.?-????????????????????????????????????????\n",
+        // Allow users of Nix to build with a dirty (uncommitted) revision
+        // but disallow this for release builds so that we can avoid releasing such a version
+        #[cfg(debug_assertions)]
```

Comment:
I think I'd just allow this unconditionally. If possible, I'd also add a comment about when this case is triggered (is it only when `--impure` is passed to the `nix build` command?)

I don't think this is a good way to ensure that "we don't release such a version", since tests are usually run on a debug build.